Maxis/wwwroot/map/shared/map.service.ts

```diff
@@ -30,27 +36,65 @@ export class MapService {
             .catch((error: any) => Observable.throw(error.json().error || 'Server error'));
 
     }
+    /*getload is for searchpoint and range*/
+    getload(point: any): Observable<any> {
+        this._point = point;
+
+        let body = JSON.stringify(point);
+
+        let headers = new Headers({ 'Content-Type': 'application/json' });
+        let options = new RequestOptions({ headers: headers });
+
+        return this._http.post(this._mapLRDUrl, body, options)
+            .map(this.extractData)
+            .catch(this.handleError);
+    }
 
     getLRD(value: any): Observable<any> {
         let point = this._point;
+      //  point = "POINT (101.743026080221 2.99700230112665)";
         this._range = value.value;
-        let body = JSON.stringify({ SearchPoint: this._point, Range: this._range });
+        let body = JSON.stringify({ SearchPoint: point, Range: this._range });
         return this._http.post(this._mapLRDUrl, body, this._requestOption)
             .map(this.extractData)
             .catch(this.handleError);
     }
+    getStruct(value: any): Observable<any> {
+        let point = this._point;

```

Comment:
do not declare variable to hold duplicate values unnecessarily as it consumes memory, use this._point instead of point.